nixos/modules/image/repart-image.nix

```diff
@@ -92,39 +88,65 @@ runCommand imageFileBasename
 
   env = mkfsEnv;
 
+  inherit partitionsJSON definitionsDirectory;
+
+  # relative path to the repart definitions that are read by systemd-repart
+  finalRepartDefinitions = "repart.d";
+
   systemdRepartFlags = [
     "--dry-run=no"
     "--empty=create"
     "--size=auto"
     "--seed=${seed}"
-    "--definitions=${amendedRepartDefinitions}"
+    "--definitions=${finalAttrs.finalRepartDefinitions}"
     "--split=${lib.boolToString split}"
     "--json=pretty"
   ] ++ lib.optionals (sectorSize != null) [
     "--sector-size=${toString sectorSize}"
   ];
 
-  passthru = {
-    inherit amendRepartDefinitions amendedRepartDefinitions;
-  };
-} ''
-  mkdir -p $out
-  cd $out
-
-  echo "Building image with systemd-repart..."
-  unshare --map-root-user fakeroot systemd-repart \
-    ''${systemdRepartFlags[@]} \
-    ${imageFileBasename}.raw \
-    | tee repart-output.json
-
-  # Compression is implemented in the same derivation as opposed to in a
-  # separate derivation to allow users to save disk space. Disk images are
-  # already very space intensive so we want to allow users to mitigate this.
```

Comment:
Do you think this comment is superfluous, or did you just miss it?

Reply:
My mistake, I dropped this when I was copy-pasting from runCommand to mkDerivation